feat: improve multi-language font support in GIF generation

weibing-bab opened the pull request:
﻿- Add system-specific Unicode font support for Windows/macOS/Linux
- Prioritize native system fonts with good Unicode coverage:
- Windows: Microsoft YaHei, Segoe UI
- macOS: PingFang, Apple Color Emoji
- Linux: DejaVuSans
- Add common fallback fonts for better compatibility
- Add debug logging for font loading
- Improve error handling with informative warnings
﻿
This change ensures proper text rendering for multiple languages 
in generated GIFs without requiring additional font installations.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/browser-use/browser-use?pullRequest=711) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/browser-use/browser-use?pullRequest=711) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/browser-use/browser-use?pullRequest=711) it.</sub>

weibing-bab commented:
fix: add multi-language support for GIF text rendering

When running the following script with Chinese text, encountered character 
encoding issues in the generated GIF:

```python
import os
import sys

sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

import asyncio
from langchain_openai import ChatOpenAI
from browser_use import Agent
from dotenv import load_dotenv

# Load environment variables
load_dotenv()
if not os.getenv('OPENAI_API_KEY'):
    raise ValueError('OPENAI_API_KEY is not set. Please add it to your environment variables.')

async def main():
    llm = ChatOpenAI(model='gpt-4o-mini')
    agent = Agent(
		task="打开百度",
		llm=llm,
	)
    await agent.run()
    input('Press Enter to exit')

if __name__ == "__main__":
    asyncio.run(main())
```
Changes made:
- Added system-specific Unicode font support for Windows/macOS/Linux
- Prioritized native system fonts with CJK support
- Added Arial Unicode MS and other fallback fonts
- Improved font loading error handling with debug logs

This ensures proper text rendering for multiple languages (CJK, etc.) 
in generated GIFs while maintaining compatibility with existing font options.

![agent_history](https://github.com/user-attachments/assets/f34ffe99-26de-48b1-862c-b4bf14ec0c78)

MagMueller commented:
Thank you very much
